include/spblas/vendor/onemkl_sycl/spmv_impl.hpp

```diff
@@ -24,28 +25,53 @@
 
 namespace spblas {
 
+class spmv_state_t {
+public:
+  spmv_state_t() : spmv_state_t(mkl::mkl_allocator<char>{}) {}
+
+  spmv_state_t(sycl::queue* q) : spmv_state_t(mkl::mkl_allocator<char>{q}) {}
+
+  spmv_state_t(mkl::mkl_allocator<char> alloc) : alloc_(alloc) {}
+
+  sycl::queue* queue() {
+    return alloc_.queue();
+  }
+
+private:
+  mkl::mkl_allocator<char> alloc_;
+};
```

Comment:
should we switch to using `uint8_t`  instead of `char` as a 1 byte = 8 bits intent ?

Reply:
again, I'm not sure we want a `template T` on the allocator, but rather on allocate the allocate/deallocate member functions, right ?